packages/visual-editor/src/components/puck/ExampleRepeatableItem.tsx

```diff
@@ -0,0 +1,168 @@
+  text: YextEntityField<string>;
+  isCollection?: boolean;
+  image: YextEntityField<ImageType>;
+};
+
+const ExampleRepeatableItemFields: Fields<ExampleRepeatableItemProps> = {
+  text: YextCollectionSubfieldSelector<any, string>({
+    label: "Text",
+    isCollection: false,
+    filter: {
+      types: ["type.string"],
+    },
+  }),
+  image: YextCollectionSubfieldSelector<any, ImageType>({
+    label: "Image",
+    isCollection: false,
+    filter: {
+      types: ["type.image"],
+    },
+  }),
+};
+
+const ExampleRepeatableItem = (props: ExampleRepeatableItemProps) => {
+  const { text, image, isCollection } = props;
+  const document = useDocument();
+
+  const collectionContext = useCollectionContext();
+
+  // If not in a collection, render a single card with normal entity fields
+  if (!isCollection || !collectionContext) {
+    const resolvedText = resolveYextEntityField(document, text);
+    const resolvedImage = resolveYextEntityField(document, image);
+
+    return (
+      <div>
+        <ExampleRepeatableItemCard image={resolvedImage} text={resolvedText} />
+      </div>
+    );
+  }
+
+  const { parentEntityField, limit } = collectionContext;
+  // If in a collection, get and resolve the parent
+  const resolvedParent = resolveYextEntityField(document, parentEntityField);
+
+  // Return one card with resolved subfields for each item in the parent
+  return (
+    <div className="flex gap-4 max-w-pageSection-maxWidth">
+      {resolvedParent?.slice(0, limit).map((item, i) => {
+        const resolvedImage = resolveYextSubfield(item, image);
+        const resolvedText = resolveYextSubfield(item, text);
+        return (
+          <ExampleRepeatableItemCard
+            key={i}
+            image={resolvedImage}
+            text={resolvedText}
+          />
+        );
+      })}
+    </div>
+  );
+};
+
+type ExampleRepeatableItemCardProps = {
+  image?: ImageType | ComplexImageType;
+  text?: string;
+};
+// Example Card -- would replace with a nice React component
+const ExampleRepeatableItemCard = (props: ExampleRepeatableItemCardProps) => {
+  const { text } = props;
+
+  // Handle ImageType or ComplexImageType
+  // Should clean up the image handling in a real card
+  let image;
+  if (
+    props.image &&
+    typeof props.image === "object" &&
+    "image" in props.image
+  ) {
+    image = props.image.image;
+  } else if (props.image) {
+    image = props.image;
+  }
+
+  return (
+    <div className="w-[200px] h-[200px] border-dashed border-black border-4 flex flex-col items-center justify-center">
+      {image?.url && (
+        <div className="mx-8">
+          <Image image={image} layout="fixed" width={100} aspectRatio={1} />
+        </div>
+      )}
+      {text && <p>{text}</p>}
+    </div>
+  );
+};
+
+export const ExampleRepeatableItemComponent: ComponentConfig<ExampleRepeatableItemProps> =
+  {
+    label: "Repeatable Card",
+    fields: ExampleRepeatableItemFields,
+    resolveFields: (data, params) => {
```

Comment:
I think if the parent changes between constant field/entity mapping we should clear out the children. We'd have to ask Aaron. I was able to set collection to constant value, add a repeatable item and configure it, then change the collection to entity field, then I had 4 items in the collection: my first constant one followed by the auto mapped items. Unclear if this is desirable or not.